core/src/main/scala/fr/gospeak/core/services/storage/ContactRepo.scala

```diff
@@ -0,0 +1,22 @@
+package fr.gospeak.core.services.storage
+
+import java.time.Instant
+
+import cats.effect.IO
+import fr.gospeak.core.domain._
+import fr.gospeak.libs.scalautils.domain.{Done, EmailAddress, Page}
+
+trait ContactRepo {
+
+  def list(partner: Partner.Slug): IO[Seq[Contact]]
+
+  def create(data: Contact.Data, by: User.Id, now: Instant): IO[Contact]
+
+  def list(partner: Partner.Slug, params: Page.Params): IO[Page[Contact]]
+
+  def find(partner: Partner.Slug, id: Contact.Id): IO[Option[(Partner, Contact)]]
```

Comment:
Not sure if it worth it returning the tuple `(Partner, Contact)`, it saves a call from the controller but mixes the concerns :(

Reply:
We need to discuss this point. The partner infos are used on the Ctrl to compute URL.